samples/multi-agent/src/main/java/demo/multiagent/application/agents/WeatherServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package demo.multiagent.application.agents;
+
+import akka.javasdk.http.HttpClient;
+import akka.javasdk.http.HttpClientProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+
+// tag::function-tool[]
+public class WeatherServiceImpl implements WeatherService {
```

Comment:
How about renaming it to `HttpWeatherService` or `WeatherApiService` to avoid `Impl` postfix?